libraries/chain/controller.cpp

```diff
@@ -233,8 +148,7 @@ struct controller_impl {
    std::function<void()>               shutdown;
    chainbase::database                 db;
    chainbase::database                 reversible_blocks; ///< a special database to persist blocks that have successfully been applied but are still reversible
-   b1::chain_kv::database              kv_database;
-   b1::chain_kv::undo_stack            kv_undo_stack;
+   combined_database                   kv_db;
```

Comment:
I'm confused with this too.  It seems as if you don't need `db` anymore and would just use `combined_database db;`. I'm sure why 1) combined_database is only being used for the rocksdb side and 2) why we still have both, that is the point of the combined_database. Or am I missing something?

Reply:
We have both because the native data (like account_object, etc) is still in chainbase, so both need to be managed when we have a chain_kv.